### PR TITLE
Add fix for ID.me

### PR DESCRIPTION
**Pull request by johnmcmahon**
How does this look?

### CODE REVIEW

File: bfapi/service/users.py [hunk 1]
```diff
@@ -173,14 +173,23 @@ def _fetch_geoaxis_profile(access_token: str):
...
 
     user_id = profile.get('DN')
+    uid = profile.get('UID')
+    firstname = profile.get('firstname')
+    lastname = profile.get('lastname')
     if not user_id:
-        log.error('Geoaxis response missing `DN`: %s', response.text)
-        raise InvalidGeoaxisResponse('missing `DN`', response.text)
+        if lastname and firstname and uid:
+            user_id = 'O=Beachfront,OU=ID.me,CN={}.{},UID={}'.format(lastname, firstname, uid)
```
Comment (russellorf) [inline bfapi/service/users.py]: This seems reversed. I think we want it to resemble:

UID=russell.f.orf.ctr@nga.mil, CN=ORF.RUSSELL, OU=ID.me, O=Beachfront

File: bfapi/service/users.py [hunk 2]
```diff
@@ -173,14 +173,23 @@ def _fetch_geoaxis_profile(access_token: str):
...
+        else:
+            log.error('Geoaxis response missing important values: %s', response.text)
+            raise InvalidGeoaxisResponse('missing important values', response.text)
```
Comment (russellorf) [inline bfapi/service/users.py]: Consider adding a debug log for when user_id is populated. Try to always log every possible condition for easier troubleshooting.

File: bfapi/service/users.py [hunk 3]
```diff
@@ -173,14 +173,23 @@ def _fetch_geoaxis_profile(access_token: str):
...
+    uid = profile.get('UID')
+    firstname = profile.get('firstname')
+    lastname = profile.get('lastname')
     if not user_id:
-        log.error('Geoaxis response missing `DN`: %s', response.text)
-        raise InvalidGeoaxisResponse('missing `DN`', response.text)
+        if lastname and firstname and uid:
+            user_id = 'O=Beachfront,OU=ID.me,CN={}.{},UID={}'.format(lastname, firstname, uid)
+        else:
+            log.error('Geoaxis response missing important values: %s', response.text)
```
Comment (russellorf) [inline bfapi/service/users.py]: Consider updating the error text to be specific to the values that are missing, for easier troubleshooting.

File: bfapi/service/users.py [hunk 4]
```diff
@@ -173,14 +173,23 @@ def _fetch_geoaxis_profile(access_token: str):
...
+            raise InvalidGeoaxisResponse('missing important values', response.text)
 
     name = profile.get('commonname')
     if not name:
-        log.error('Geoaxis response missing `commonname`: %s', response.text)
-        raise InvalidGeoaxisResponse('missing `commonname`', response.text)
+        if lastname and firstname:
+            name = '{}.{}'.format(lastname, firstname)
+        else:
+            log.error('Geoaxis response missing name values: %s', response.text)
```
Comment (russellorf) [inline bfapi/service/users.py]: Consider updating the error text to be specific to the values that are missing, for easier troubleshooting.

File: bfapi/service/users.py [hunk 5]
```diff
@@ -173,14 +173,23 @@ def _fetch_geoaxis_profile(access_token: str):
...
+            raise InvalidGeoaxisResponse('missing name values', response.text)
```
Comment (russellorf) [inline bfapi/service/users.py]: Consider adding a debug log for when name is populated. Try to always log every possible condition for easier troubleshooting.

File: bfapi/service/users.py (hunk 1 above)
Comment (johnmcmahon) [reply bfapi/service/users.py]: ah ok, i wasn't sure if order mattered here.  I'll reorder now

File: bfapi/service/users.py (hunk 3 above)
Comment (johnmcmahon) [reply bfapi/service/users.py]: I assumed response.text would contain all the values that actually came back but I guess that might not be true.

**Pull request merged**